### PR TITLE
WIP: Lammpsdump velocities (and other fields)

### PR DESCRIPTION
**Pull request by schlaicha**
Following #3360 (thanks for the great work!) I was implementing a first draft of a parser for the velocities.
Before adding tests I quickly wanted to hear if the way of implementing agrees with your concepts.
Other fields, like e.g. forces, could be added straightforward.

Fixes #

Changes made in this Pull Request:
 - allow LAMMPSDUMP parser to read velocities

PR Checklist
------------
 - [ ] Tests?
 - [ x] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/LAMMPS.py [hunk 1]
```diff
@@ -502,7 +504,7 @@ def _reopen(self):
         self.close()
         self._file = util.anyopen(self.filename)
         self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
-        self.ts.frame = -1
```
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/LAMMPS.py:505]: this is deliberately -1 as it gets incremented in the loading process

**Review by richardjgowers — COMMENTED** [review]
Hi @schlaicha thanks for starting this.  Yes this looks like the correct approach to get this working.

File: package/MDAnalysis/coordinates/LAMMPS.py [hunk 2]
```diff
@@ -606,15 +608,32 @@ def _read_next_timestep(self):
 
         coord_cols = convention_to_col_ix[self.lammps_coordinate_convention]
 
+        # pass possible additional fields from LAMMPS dump
+        # pass velocities
+        velocity_col_names = ["vx", "vy", "vz"]
```
Comment (hmacdope) [inline package/MDAnalysis/coordinates/LAMMPS.py:613]: I would make this a class attribute, much like `self._conventions` is, as the data is not mutable or instance dependent. :)

File: package/MDAnalysis/coordinates/LAMMPS.py [hunk 3]
```diff
@@ -606,15 +608,32 @@ def _read_next_timestep(self):
...
+        if any(item in attrs for item in velocity_col_names):
+            for cv_name, cv_col_names in enumerate(velocity_col_names):
+                try:
+                    velocity_cols = [attr_to_col_ix[x] for x in velocity_col_names]
+                    ts.has_velocities = True
```
Comment (hmacdope) [inline package/MDAnalysis/coordinates/LAMMPS.py:618]: This works, but I think we only want to set `ts.has_velocities = True` on exit from this loop, slightly cleaner IMO.

File: package/MDAnalysis/coordinates/LAMMPS.py [hunk 4]
```diff
@@ -606,15 +608,32 @@ def _read_next_timestep(self):
...
             ts.positions[i] = [fields[dim] for dim in coord_cols]
+            if ts.has_velocities:
+                ts.velocities[i] = [fields[dim] for dim in velocity_cols]
 
         order = np.argsort(indices)
         ts.positions = ts.positions[order]
+        if ts.has_velocities:
+            ts.velocities = ts.velocities[order]
+            # LAMMPS reader only supports real units
+            ts.velocities *= units.speedUnit_factor['Angstrom/femtosecond']
```
Comment (hmacdope) [inline package/MDAnalysis/coordinates/LAMMPS.py:636]: Query @richardjgowers on units? I'm no lammps expert, what are the range of possible units?

**Review by hmacdope — CHANGES_REQUESTED** [review]
Thanks so much for working on this! A few comments, but generally looking good. 👍

**Comment by orbeckst** [comment]
I am going to close and re-open to trigger CI.

**Comment by orbeckst** [comment]
@schlaicha are you still interested in completing the PR?

**Comment by schlaicha** [comment]
> @schlaicha are you still interested in completing the PR?

Sorry for the long delay, in fact I forgot about that one.

My collaborator @pstaerk started implementing this in a more general way, I believe, and thus I would suggest that we try to merge this with #3608

We will come back to this after having a first comment on #3608

**Comment by hejamu** [comment]
@schlaicha the changes in this PR have been superseded by #3844.

**Comment by schlaicha** [comment]
Thanks @hejamu for reminding me about this open PR. Closing as it is outdated.

**Pull request closed without merging**